src/Xunit.StaFact.Desktop/Sdk/DispatcherSynchronizationContextAdapter.cs

```diff
@@ -41,5 +41,11 @@ internal override void Run(Func<Task> work)
             this.PumpTill(task);
             task.GetAwaiter().GetResult();
         }
+
+        internal override void Cleanup()
+        {
+            Dispatcher.CurrentDispatcher.InvokeShutdown();
```

Comment:
I just realized that this code is probably wrong depending on which thread it gets run on.[ InvokeShutdown is async ](https://msdn.microsoft.com/en-us/library/system.windows.threading.dispatcher.invokeshutdown(v=vs.110).aspx) so depending on how fast XUnit tears everything down and how fast the dispatcher event queue can be flushed you might still see the exception. 

One way to synchronise is to use a TaskCompletionSource

            var tc = new TaskCompletionSource<bool>();
            var h = (EventHandler)((sender,e) => tc.SetResult(true));
            Dispatcher.CurrentDispatcher.ShutdownFinished += h;
            Dispatcher.CurrentDispatcher.InvokeShutdown();
            tc.Task.Wait();
            Dispatcher.CurrentDispatcher.ShutdownFinished -= h;

It's ugly and I think it will deadlock if run on the dispatcher thread and there are items in the dispatcher queue. But some technique for waiting for the dispatcher to shutdown is required I think.